src/Symfony/Component/HttpClient/Response/ResponseTrait.php

```diff
@@ -235,6 +235,15 @@ private static function addResponseHeaders(array $responseHeaders, array &$info,
         }
     }
 
+    private static function handleContentType(&$content, string $contentType)
+    {
+        $contentType = substr($contentType, 0, strpos($contentType, ';'));
+        if (1 === preg_match('/(text\/[a-z]+|[a-z]+\/(json|xml)$)/', $contentType) && null === $content) {
```

Comment:
instead of hardcoding the check, we should make it configurable:
when option "buffer" is set to a string, we could consider it a regexp that the content-type should match to enable buffering.
note that we don't use the `1 === ` part so it should be removed

Reply:
Provided a check to the buffer option if it's a string and removed the `1 ===` part